lumigator/python/mzai/sdk/lm_datasets.py

```diff
@@ -0,0 +1,91 @@
+"""
+Dataset SDK
+
+Provides a class to manipulate datasets in Lumigator.
+"""
+
+from schemas.datasets import DatasetFormat, DatasetResponse, DatasetDownloadResponse
+from schemas.extras import ListingResponse
+
+from uuid import UUID
+from http import HTTPMethod
+
+from io import IOBase
+from client import ApiClient
+
+
+class Datasets:
+    DATASETS_ROUTE = "datasets"
+
+    def __init__(self, c: ApiClient):
+        self.client = c
+
+    def get_datasets(self) -> ListingResponse[DatasetResponse]:
+        """Returns information on all datasets.
+        Returns:
+            ListingResponse[DatasetResponse]: all existing datasets.
+        """
+        response = self.client.get_response(self.DATASETS_ROUTE)
+
+        if not response:
+            return []
+
+        return [DatasetResponse(**args) for args in response.json()]
+
+    def get_dataset(self, id: str) -> DatasetResponse:
+        """Returns information on a specific dataset.
+        Args:
+            id (str): the id of the dataset to retrieve
+        Returns:
+            DatasetResponse: the dataset information for the provided id.
+        """
+        UUID(id)
```

Comment:
we likely don't want to reformat this as a uuid since it will get sent as a string to the response anyway and in the current implementation it's not assigned, aka `id = UUID(id)`

Reply:
The idea was to perform client side validation, but I guess a much more logical move is to replace `id: str` with `id: UUID` and force the user to provide something that is really a UUID. Wdyt?

Reply:
sure, sounds good.